source/MicroBitCompass.cpp

```diff
@@ -31,320 +31,55 @@ DEALINGS IN THE SOFTWARE.
 #include "MicroBitError.h"
 #include "LSM303Magnetometer.h"
 
-Compass* MicroBitCompass::driver;
+using namespace codal;
 
-/**
- * Constructor.
- * Create a software abstraction of an e-compass.
- *
- * @param id the unique EventModel id of this component. Defaults to: MICROBIT_ID_COMPASS
- * @param coordinateSpace the orientation of the sensor. Defaults to: SIMPLE_CARTESIAN
- *
- */
-MicroBitCompass::MicroBitCompass(MicroBitI2C &i2c, uint16_t id) : Compass(MicroBitAccelerometer::coordinateSpace)
-{
-    autoDetect(i2c);
-}
-
-/**
- * Device autodetection. Scans the given I2C bus for supported accelerometer devices.
- * if found, constructs an appropriate driver and returns it.
- *
- * @param i2c the bus to scan. 
- * @param id the unique EventModel id of this component. Defaults to: MICROBIT_ID_ACCELEROMETER
- *
- */
-Compass& MicroBitCompass::autoDetect(MicroBitI2C &i2c)
-{
-    /*
-     * In essence, the LSM needs at least 6.4ms from power-up before we can use it.
-     * https://github.com/microbit-foundation/codal-microbit/issues/33
-     */
-    target_wait(10);
```

Comment:
If the `MicroBitAccelerometer::autoDetect()` already adds a wait, there is no need to wait here as well, which why it was removed.